### PR TITLE
Feature/add user profile logic,  some adjustments

### PR DESCRIPTION
**Pull request by tanyushanel**
image

### CODE REVIEW

File: exadel-trainee-app/src/app/models/angular-material-common.module.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ import { MAT_RADIO_DEFAULT_OPTIONS, MatRadioModule } from '@angular/material/rad
     MatIconModule,
     MatCardModule,
     MatDialogModule,
+    // MatTableDataSource,
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/angular-material-common.module.ts:33]: please remove

File: exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.scss [hunk 2]
```diff
@@ -1,8 +1,13 @@
 .content {
   position: relative;
+  height: auto;
   white-space: nowrap;
 }
 
+.mat-dialog-content {
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.scss:7]: please remove

File: exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts [hunk 3]
```diff
@@ -13,9 +13,9 @@ import { Test } from '../../interfaces/test';
 export class UserProfileComponent implements OnInit {
   results!: Test[];
 
-  levels = Level;
+  levels = [...Object.values(Level)];
 
-  selectedLevel = Level.beginner;
+  selectedLevel!: Level;
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts:18]: lets not use ! operator. 
If you expect variable to be possible undefined use **selectedLevel: Level | undefined**

File: exadel-trainee-app/src/constants/data-constants.ts [hunk 4]
```diff
@@ -16,3 +18,8 @@ export enum UserRole {
   Coach = 'Coach',
   Administrator = 'Administrator',
 }
+
+export enum Priority {
+  high = 1,
```
Comment (RomPavel) [inline exadel-trainee-app/src/constants/data-constants.ts:23]: I see usually enum properties starts with capital letter, lets stick with one style

File: exadel-trainee-app/src/constants/mock-test-data.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import { Test } from 'src/app/interfaces/test';
-import { Level } from 'src/constants/data-constants';
+import { Level, Priority } from 'src/constants/data-constants';
 
 export const results: Test[] = [
```
Comment (RomPavel) [inline exadel-trainee-app/src/constants/mock-test-data.ts:4]: Its better to name mocks like MOCK_TEST_RESULTS
It also pretty huge, we usually try to randomly generate such mocks, see other mock files

**Review by RomPavel — APPROVED** [review]

File: exadel-trainee-app/src/app/components/user-results-table/user-results-table.component.html [hunk 6]
```diff
@@ -70,3 +71,6 @@ <h3 class="table-caption">Your recent results</h3>
   ></tr>
   <tr mat-row *matRowDef="let row; columns: ['expandedDetail']" class="detail-row"></tr>
 </table>
+
+<mat-paginator [length]="50" [pageSize]="5" [pageSizeOptions]="[5]" showFirstLastButtons>
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/components/user-results-table/user-results-table.component.html:75]: We also discussed that its better to enable **showFirstLastButtons** once for all tables

**Review by Inunor — APPROVED** [review]

**Pull request merged**